interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -36,226 +36,151 @@ angular.module('adage.gene.searchMany', [
   ;
 })
 
-.factory('SearchResults', ['$rootScope', 'Gene', function($rootScope, Gene) {
-  var queries = [];
-  var searchResults = {};
-  var searchSuccess = null;
 
+// Directive for whole gene search form
+.directive('geneSearchPanel', [function() {
   return {
-    getQueries: function() {
-      return queries;
-    },
-    getQueryResults: function(query) {
-      return searchResults[query];
-    },
-    getSearchResults: function() {
-      return searchResults;
-    },
-    getSearchSuccess: function() {
-      return searchSuccess;
-    },
-    remove: function(query) { // Remove a query and its associated result
-      queries = queries.filter(function(el) {
-        return el !== query;
-      });
-      delete searchResults[query];
-      $rootScope.$broadcast('results.update');
-    },
-    clear: function() { // Clear the service
-      queries = [];
-      searchResults = {};
-      $rootScope.$broadcast('results.update');
-    },
-    size: function() {
-      return queries.length;
+    controller: ['$scope', function($scope) {
```

Comment:
Here you can use either a controller or a link function to set variables in the directive's scope. The Angular book you have (**AngularJS: Up and Running**) has a good discussion on which one should be used under certain scenario (see **page 216**). Since you are not doing any inter-directive communication here, link function is probably a better choice.